QuizApp/Services/ServiceUtils/Router.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  Router.swift
+//  QuizApp
+//
+//  Created by Luka Namačinski on 13.05.2021..
+//
+
+import Foundation
+
+enum Router {
+    case login(username: String, password: String)
+    case getQuizzes
+    case postQuizResult(result: QuizResult)
+    case getLeaderboard(id: Int)
+
+    var scheme: String {
+        switch self {
+        default:
+            return "https"
+        }
+    }
+    
+    var host: String {
+        switch self {
+        default:
+            return "iosquiz.herokuapp.com"
+        }
+    }
+    
+    var path: String {
+        switch self {
+        case .login:
+            return "/api/session"
+        case .getQuizzes:
+            return "/api/quizzes"
+        case .postQuizResult:
+            return "/api/result"
+        case .getLeaderboard:
+            return "/api/score"
+        }
+    }
+    
+    var queryItems: [URLQueryItem]? {
+        switch self {
+        case .login(let username, let password):
+            return [URLQueryItem(name: "username", value: username),
+                    URLQueryItem(name: "password", value: password)]
+        case .getLeaderboard(let quizID):
+            return [URLQueryItem(name: "quiz_id", value: String(quizID))]
+        default:
+            return nil
+        }
+    }
+    
+    var method: String {
+        switch self {
+        case .login, .postQuizResult:
+            return "POST"
+        case .getQuizzes, .getLeaderboard:
+            return "GET"
+        }
+    }
+    
+    var body: Data? {
+        switch self {
+        case .postQuizResult(let quizResult):
+            let data = try! JSONEncoder().encode(quizResult)
+            return data
+        default:
```

Comment:
default je u redu koristiti, ali uvijek ti je bolje imati jedan `case` sa ostalim slučajevima zbog toga što kada dodaš novi `case` u enum, ne sjetiš se uvijek svih mjesta gdje si taj enum koristio u `switch case` naredbi.